Add challenges (#1)
unikraft-upb/USoC23-registration: 01-build-system/README.md

# 01-build-system

## Challenge Description

You are given the following file structure:

```console
.
|-- key.c
|-- main.c
`-- some
    `-- far
        `-- away
            |-- include
            |   `-- myheader.h
            `-- src
                `-- values.c
```

> You are asked to create a `Makefile` that will be used to build everything into an executable file.

Reviewer: ```suggestion
Your task is to create a `Makefile` that will be used to build everything into an executable file.
```

After everything is done, if you run the executable you will find a flag:

```console
$ ./main
[...]
USoC{...}
```

> Copy that flag and paste it in the [registration form](TODO), along with the `Makefile` you created.

Reviewer: Maybe they should create a zip archive in which they will upload multiple files?
